src/jdk.jpackage/share/classes/jdk/jpackage/internal/CfgFile.java

```diff
@@ -137,12 +159,47 @@ private ApplicationLayout createAppCfgLayout() {
         return appCfgLayout;
     }
 
+    private String getDefaultSearchPath(String name) {
+        if (Platform.isWindows()) {
+            Path pf = ("x86".equals(System.getProperty("os.arch")) ?
+                   getWinSystemDir("ProgramFiles(x86)","\\Program Files (x86)") :
+                   getWinSystemDir("ProgramFiles", "\\Program Files"));
+
+            return pf.resolve("Java").toString() +
+                    ((name == null) ? "" : ("," + pf.resolve(name).toString()));
+        } else if (Platform.isMac()) {
+            return "/Library/Internet Plug-ins/JavaAppletPlugin.plugin" + ","
+                    + "/Library/Java/JavaVirtualMachines";
+        } else {
+            return "/usr/lib/jvm" + "," + "/usr/lib/jdk" ;
+        }
+    }
```

Comment:
I'd use $DEFAULTSERACHPATH variable to hide system-specific paths where to look for runtimes and get it expanded like $APPDIR in app launcher. Values of environment variables on a system where the app will be executed and where it was bundles are unlikely to match.